### PR TITLE
Login and Signup Endpoints

### PR DESCRIPTION
**Pull request by Mistaweng**

### CODE REVIEW

**Review by Taiwopeter-babs — APPROVED** [review]
Excellent work! @Mistaweng. The  `protectedRoute` method in the controller functions as a middleware since it checks for the cookie before accessing the database. It would be much less debt if it verifies the access and refresh tokens for protected routes and the login endpoint.

File: src/middlewares/token.js
Comment (VivianDee) [inline src/middlewares/token.js:1]: I think there is a mistake in the createAccessToken function. It should've been 
const token = await sign({ email }, accessTokenSecret, { expiresIn: '1d' });


secretKey is not defined.

Comment (Mistaweng) [reply src/middlewares/token.js:1]: Right, i forgot to change the variable name from sectretkey to accessTokenSecret. Thank you

**Comment by Mistaweng** [comment]
> Excellent work! @Mistaweng. The `protectedRoute` method in the controller functions as a middleware since it checks for the cookie before accessing the database. It would be much less debt if it verifies the access and refresh tokens for protected routes and the login endpoint.

So, true @Taiwopeter-babs, i'll work on modifying it. Thank you

**Pull request merged**